exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -178,15 +179,25 @@ func routeRecord(
 		}
 	}
 
-	// Only use receiver-based routing if dataset is not specified.
+	// Only use extension or receiver-based routing if dataset is not specified.
 	if !datasetExists {
-		// Receiver-based routing
-		// For example, hostmetricsreceiver (or hostmetricsreceiver.otel in the OTel output mode)
-		// for the scope name
-		// github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/cpuscraper
-		if submatch := receiverRegex.FindStringSubmatch(scope.Name()); len(submatch) > 0 {
-			receiverName := submatch[1]
-			dataset = receiverName
+		// Extension-based routing
+		// For example, awslogsencodingextension sets scope attributes according to log types
+		// We should probably extend this to support a common attributes pattern (`*.format`?), but for now
+		// let's just make sure it works for awslogsencodingextension.
+		if format, ok := scope.Attributes().Get("awslogs_encoding.format"); ok {
```

Comment:
Lets keep it generic, and get rid of the AWS prefix. We can make the changes on the encoding extension. 

```suggestion
		if format, ok := scope.Attributes().Get("encoding.format"); ok {
```

Reply:
See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/42899